### PR TITLE
fix: quick fix for undefined cves values

### PR DESCRIPTION
**Pull request by helio-frota**
Related to https://github.com/nodeshift/nodejs-reference-architecture/issues/94

### CODE REVIEW

**Comment by helio-frota** [comment]
I'm not 100% sure that will work but by commenting `advisoryData` section on the unit tests we can see that the test passes, let's see what happen with the CI :+1:

**Comment by richardlau** [comment]
I'm a little concerned this is masking some other issue. For example, trying to run `npcheck` with nodejs-reference-architecture's `npcheck.json` doesn't complete and gets a 403 error which I suspect may be rate limiting/DDOS protection kicking in.

**Comment by helio-frota** [comment]
@richardlau I agree with that, I saw the same and that I decided to be kind of incisive/pragmatic with this fix.

**Comment by helio-frota** [comment]
we need to be able to replicate errors locally without 403 so yeah maybe is good to keep this PR open for a bit

**Comment by aalykiot** [comment]
I'll try to take a look at this issue/PR as soon as I manage to find some time (and energy).

**Comment by aalykiot** [comment]
I'll probably take a look this weekend which I'll have plenty of free time 🙂

**Comment by helio-frota** [comment]
@richardlau @aalykiot PR update with more bits fixed I think now is more save to merge.

This is the result for the odbc module:

```
Running checks on odbc

Downloading module dependencies...

odbc@2.4.1 has been installed.

Checking if it's deprecated on NPM                                PASS 
Checking if github repository is archived                         PASS 

Checking top-level license                                        PASS 
Checking license of @mapbox/node-pre-gyp@1.0.6                    PASS 
Checking license of abbrev@1.1.1                                  PASS 
Checking license of agent-base@6.0.2                              PASS 
Checking license of ansi-regex@3.0.0                              PASS 
Checking license of aproba@2.0.0                                  PASS 
Checking license of are-we-there-yet@2.0.0                        PASS 
Checking license of async@3.2.2                                   PASS 
Checking license of balanced-match@1.0.2                          PASS 
Checking license of brace-expansion@1.1.11                        PASS 
Checking license of chownr@2.0.0                                  PASS 
Checking license of color-support@1.1.3                           PASS 
Checking license of concat-map@0.0.1                              PASS 
Checking license of console-control-strings@1.1.0                 PASS 
Checking license of debug@4.3.2                                   PASS 
Checking license of delegates@1.0.0                               PASS 
Checking license of detect-libc@1.0.3                             PASS 
Checking license of fs-minipass@2.1.0                             PASS 
Checking license of fs.realpath@1.0.0                             PASS 
Checking license of gauge@3.0.1                                   PASS 
Checking license of glob@7.2.0                                    PASS 
Checking license of has-unicode@2.0.1                             PASS 
Checking license of https-proxy-agent@5.0.0                       PASS 
Checking license of inflight@1.0.6                                PASS 
Checking license of inherits@2.0.4                                PASS 
Checking license of is-fullwidth-code-point@2.0.0                 PASS 
Checking license of lru-cache@6.0.0                               PASS 
Checking license of make-dir@3.1.0                                PASS 
Checking license of minimatch@3.0.4                               PASS 
Checking license of minipass@3.1.5                                PASS 
Checking license of minizlib@2.1.2                                PASS 
Checking license of mkdirp@1.0.4                                  PASS 
Checking license of ms@2.1.2                                      PASS 
Checking license of node-addon-api@3.2.1                          PASS 
Checking license of node-fetch@2.6.6                              PASS 
Checking license of nopt@5.0.0                                    PASS 
Checking license of npmlog@5.0.1                                  PASS 
Checking license of object-assign@4.1.1                           PASS 
Checking license of once@1.4.0                                    PASS 
Checking license of path-is-absolute@1.0.1                        PASS 
Checking license of readable-stream@3.6.0                         PASS 
Checking license of rimraf@3.0.2                                  PASS 
Checking license of safe-buffer@5.2.1                             PASS 
Checking license of semver@6.3.0                                  PASS 
Checking license of semver@7.3.5                                  PASS 
Checking license of set-blocking@2.0.0                            PASS 
Checking license of signal-exit@3.0.5                             PASS 
Checking license of string-width@2.1.1                            PASS 
Checking license of string_decoder@1.3.0                          PASS 
Checking license of strip-ansi@4.0.0                              PASS 
Checking license of tar@6.1.11                                    PASS 
Checking license of tr46@0.0.3                                    PASS 
Checking license of util-deprecate@1.0.2                          PASS 
Checking license of webidl-conversions@3.0.1                      PASS 
Checking license of whatwg-url@5.0.0                              PASS 
Checking license of wide-align@1.1.5                              PASS 
Checking license of wrappy@1.0.2                                  PASS 
Checking license of yallist@4.0.0                                 PASS 

Checking if module has appropriate testing                        PASS 

Checking maintenance metrics                                      PASS 

Checking for LTS support                                          PASS 

Checking for TypeScript typings                                   PASS 

Checking if module is tested by community CITGM runs              WARN 

Checking for active vulnerabilities                               PASS 

NPCheck Report

(1): The module "odbc" is not tested by community CITGM runs.

problems: 1 (errors: 0 - warnings: 1)
```

File: src/plugins/audit.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ const auditPlugin = async () => {
         headers: { 'X-Spiferack': 1 }
       });
 
-      const advisory = { cve: data.advisoryData.cves[0], events: data.events };
+      const advisory = { cve: data.advisoryData?.cves[0], events: data.events };
```
Comment (richardlau) [inline src/plugins/audit.js:60]: I'm still investigating... but with `odbc`, `data` is coming back as HTML (for https://github.com/advisories/GHSA-93q8-gq69-wqmw) and not JSON.

Comment (richardlau) [reply src/plugins/audit.js:60]: Hmm something may have changed... from https://github.com/nodeshift/npcheck/issues/58#issuecomment-906448740 `curl -H 'X-Spiferack: 1' https://www.npmjs.com/advisories/1756` no longer returns the JSON in that comment,
```console
$  curl -H 'X-Spiferack: 1' https://www.npmjs.com/advisories/1756
<h1>Redirect...</h1>$
```

**Comment by richardlau** [comment]
This will make the CVE check useless -- upstream changes to the advisory endpoint we were accessing means we no longer get JSON data from it so are unable to check the CVE dates. This PR would just hide the problem and we'd never get errors/warnings.

Opened https://github.com/nodeshift/npcheck/pull/91 to change the advisory endpoint we use.

**Pull request closed without merging**